clang/test/CodeGenCXX/pr70585.cppm

```diff
@@ -0,0 +1,50 @@
+// REQUIRES: !system-windows
+
+// RUN: rm -rf %t
+// RUN: split-file %s %t
+// RUN: cd %t
+//
+// RUN: %clang_cc1 -std=c++20 %t/layer1.cppm -triple %itanium_abi_triple \
+// RUN:     -emit-module-interface -o %t/foo-layer1.pcm
+// RUN: %clang_cc1 -std=c++20 %t/layer2.cppm -triple %itanium_abi_triple  \
+// RUN:     -emit-module-interface -fmodule-file=foo:layer1=%t/foo-layer1.pcm \
+// RUN:     -o %t/foo-layer2.pcm
+// RUN: %clang_cc1 -std=c++20 %t/foo-layer1.pcm -S -emit-llvm -o - | FileCheck %t/layer1.cppm
+// RUN: %clang_cc1 -std=c++20 %t/foo-layer2.pcm -S -emit-llvm -o - \
+// RUN:     -fmodule-file=foo:layer1=%t/foo-layer1.pcm | FileCheck %t/layer2.cppm
+
+//--- layer1.cppm
+export module foo:layer1;
+struct Fruit {
+    virtual ~Fruit() = default;
+    virtual void eval() = 0;
+};
+struct Banana : public Fruit {
+    Banana() {}
+    void eval() override;
+};
+
```

Comment:
What different cases do these two classes test? Would one class be sufficient?

Reply:
This comes from the original reproducer while the base class is a pure virtual class that can't be instantiated. While it may be sufficient to reduce them to a single class, I feel slightly better to keep the form since it is not complex.

Reply:
Not complex, but does obscure the intent of the test somewhat - keeping things as reduced as possible makes the test easier to understand - having superfluous details makes it harder to read/understand the purpose of the test.

Reply:
Great suggestion. I caught another bug after I removed `Banana`: previously we would only generate vtable if it is used. And my draft to fix that is https://github.com/ChuanqiXu9/llvm-project/commit/f6a22849bb410532c60dca4453fd2b0b71da3994. It touches Sema and Serializer. So I feel it is a little bit complex even if it is small. Personally, I prefer to separate that into 2 patches later instead of combining everything in the current single patch. How do you feel about the direction? @dwblaikie @rjmccall

Reply:
I've just tried to update it here. If you feel it is too sparse, I can try to make it into 2 patches.